hms-app/src/components/admin-tools/UserForm.tsx

```diff
@@ -0,0 +1,197 @@
+import { CheckIcon } from '@modulz/radix-icons'
+import { editUser, getUserDetails } from '../../actions/UserActions'
+import { SkillPreview } from '../../common/types'
+import { getListOfSkills } from '../../actions/SkillActions'
+
+type IProps = {
+  contextID: string
+}
+
+const useStyles = createStyles((theme) => ({
+  card: {
+    backgroundColor:
+      theme.colorScheme === 'dark' ? theme.colors.dark[7] : theme.white,
+  },
+  section: {
+    borderBottom: `1px solid ${
+      theme.colorScheme === 'dark' ? theme.colors.dark[4] : theme.colors.gray[3]
+    }`,
+    paddingLeft: theme.spacing.md,
+    paddingRight: theme.spacing.md,
+    paddingBottom: theme.spacing.md,
+    textTransform: 'uppercase',
+    fontSize: theme.fontSizes.md,
+    fontWeight: 500,
+  },
+}))
+
+export default function UserForm(props: IProps) {
+  const { classes } = useStyles()
+  const { contextID } = props
+  const [isError, setIsError] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+  const [availableSkills, setAvailableSkills] = useState({
+    skills: [] as SkillPreview[],
+  })
+  const [skills, setSkills] = useState<string[]>([])
+  const [user, setUser] = useState({
+    lastName: '',
+    firstName: '',
+    emailAddress: '',
+    roles: [],
+    imageUrl: '',
+  })
+
+  const loadSelectedUser = () => {
+    getUserDetails(contextID).then(
+      (data) => {
+        setIsError(false)
+        setIsLoading(false)
+        setUser({
+          lastName: data.lastName,
+          firstName: data.firstName,
+          emailAddress: data.emailAddress,
+          roles: data.roles,
+          imageUrl: data.imageUrl,
+        })
+      },
+      () => {
+        setIsError(true)
+        setIsLoading(false)
+      }
+    )
+  }
+
+  useEffect(() => {
+    loadSelectedUser()
+    loadAvailableSkills()
+    setIsLoading(true)
+  }, [])
+
+  function handleChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+    setUser((prevUserText) => ({
+      ...prevUserText,
+      [event.target.name]: event.target.value,
+    }))
+  }
+
+  function editThisUser(event: React.MouseEvent<HTMLButtonElement>) {
+    event.preventDefault()
+    showNotification({
+      id: 'user-load',
+      loading: true,
+      title: 'User is uploading',
+      message: 'this can take a second',
+      autoClose: false,
+      disallowClose: true,
+    })
+    editUser(contextID, user, skills).then((r) =>
+      setTimeout(() => {
+        updateNotification({
+          id: 'user-load',
+          color: 'teal',
+          title: 'User was added',
+          message: 'Notification will close in 2 seconds',
+          icon: <CheckIcon />,
+          autoClose: 2000,
+        })
+      }, 3000)
+    )
+  }
+
+  function submitIsEnabled(): boolean {
+    return !!user.firstName
+  }
+
+  const loadAvailableSkills = () => {
+    getListOfSkills().then((data) => {
+      setAvailableSkills({
+        ...availableSkills,
+        skills: data.skills,
+      })
```

Comment:
I expect this won't work quite right.  availableSkills is meant to be an array not an object, but this will create an object with a field inside it called `skills` that is an string[] and another string[] with no name... all you should do really is `setAvailableSkills(data.skills)`

Reply:
we should assume in this instance that the data we get from the backend is the complete list of skills so we can just wipe out whatever data is currently stored in `availableSkills` and replace it with the data from the backend